src/main/java/com/avispl/symphony/dal/infrastructure/management/nec/navisetadministrator2se/NaViSetAdministrator2SECommunicator.java

```diff
@@ -0,0 +1,1251 @@
+			}
+		}
+	}
+
+	/**
+	 * Sends a control command to the specified device.
+	 *
+	 * @param deviceId The unique identifier of the target device.
+	 * @param code The VCP code associated with the control property.
+	 * @param name The name of the control property.
+	 * @param value The numerical value to set for the control property.
+	 * @param textValue The textual representation of the value (for error reporting).
+	 * @throws IllegalArgumentException If an error occurs during the control command execution.
+	 */
+	private void sendControlCommand(String deviceId, String code, String name, String value, String textValue) {
+		try {
+			ObjectNode body = objectMapper.createObjectNode();
+			body.put("id", deviceId);
+			body.put("vcpcode", code);
+			body.put("vcpvalue", value);
+			JsonNode response = this.doPut(NaViSetAdministrator2SECommand.CONTROL_COMMAND, (JsonNode) body, JsonNode.class);
+			if (response == null || !response.has(NaViSetAdministrator2SEConstant.DATA)) {
+				throw new IllegalArgumentException("Error setting a control on the device.");
+			}
+		} catch (Exception e) {
+			throw new IllegalArgumentException(String.format("Can't control property %s with value %s", name, textValue), e);
+		}
+	}
+
+	/**
+	 * Retrieves system information using a GET request to the specified command.
+	 * The result is stored in the aggregatorResponse field.
+	 *
+	 * @throws IllegalArgumentException If the response does not contain the expected data field
+	 * or if there is an error during the retrieval process.
+	 */
+	private void retrieveSystemInfo() {
+		try {
+			aggregatorResponse = this.doGet(NaViSetAdministrator2SECommand.SYSTEM_INFO_COMMAND, JsonNode.class);
+			if (!aggregatorResponse.has(NaViSetAdministrator2SEConstant.DATA)) {
+				throw new IllegalArgumentException("The response is not correct.");
+			}
+		} catch (Exception e) {
+			throw new IllegalArgumentException("Error while retrieve system information" + e);
+		}
+	}
+
+	/**
+	 * Populates the provided stats map with system information extracted from the aggregator response.
+	 * The mapping is based on the SystemInformation enum.
+	 *
+	 * @param stats The map to populate with system information.
+	 */
+	private void populateSystemInfo(Map<String, String> stats) {
+		for (SystemInformation item : SystemInformation.values()) {
+			String propertyName = item.getName();
+			String value = getDefaultValueForNullData(aggregatorResponse.get(NaViSetAdministrator2SEConstant.DATA).get(item.getValue()).asText());
+			stats.put(propertyName, value);
+		}
+	}
+
+	/**
+	 * Populates device details using a multithreaded approach by retrieving aggregated data for each device ID.
+	 * The number of threads is set to 8, and a thread pool is utilized to parallelize the retrieval process.
+	 * The results are printed to the console, and the thread pool is shutdown upon completion.
+	 */
+	private void populateDeviceDetails() {
+		int numberOfThreads = 8;
```

Comment:
I think, number of threads may be exposed to the configurable property, with a default value set in aggregator 

Rn Im collecting a list of things to make "default functionality" for each adapter, this may be one of them :)